frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx

```diff
@@ -0,0 +1,186 @@
+import { Select } from 'antd';
+// ** Constants
+import { QUERY_BUILDER_OPERATORS_BY_TYPES } from 'constants/queryBuilder';
+import {
+	transformFromStringToHaving,
+	transformHavingToStringValue,
+} from 'lib/query/transformQueryBuilderData';
+// ** Helpers
+import { transformStringWithPrefix } from 'lib/query/transformStringWithPrefix';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { Having } from 'types/api/queryBuilder/queryBuilderData';
+import { SelectOption } from 'types/common/select';
+
+// ** Types
+import { HavingFilterProps } from './HavingFilter.interfaces';
+
+const { Option } = Select;
+
+const initialFormValues: Having = {
+	columnName: '',
+	op: '',
+	value: '',
+};
+
+export function HavingFilter({
+	query,
+	onChange,
+}: HavingFilterProps): JSX.Element {
+	const { having } = query;
+	const [searchText, setSearchText] = useState<string>('');
+	const [options, setOptions] = useState<SelectOption<string, string>[]>([]);
+	const [localValues, setLocalValues] = useState<string[]>([]);
+	const [currentFormValue, setCurrentFormValue] = useState<Having>(
+		initialFormValues,
+	);
+
+	const numberOperators = QUERY_BUILDER_OPERATORS_BY_TYPES.number;
+
+	const aggregatorAttribute = useMemo(
+		() =>
+			transformStringWithPrefix({
+				str: query.aggregateAttribute.key,
+				prefix: query.aggregateAttribute.type || '',
+				condition: !query.aggregateAttribute.isColumn,
+			}),
+		[query],
+	);
+
+	const columnName = useMemo(
+		() => `${query.aggregateOperator.toUpperCase()}(${aggregatorAttribute})`,
+		[query, aggregatorAttribute],
+	);
+
+	const aggregatorOptions: SelectOption<string, string>[] = useMemo(
+		() => [{ label: columnName, value: columnName }],
+		[columnName],
+	);
+
+	const generateOptions = useCallback(
+		(search: string): void => {
+			const [aggregator = '', op = '', value = ''] = search.split(' ');
+			let newOptions: SelectOption<string, string>[] = [];
+
+			const isAggregatorExist = columnName
+				.toLowerCase()
+				.includes(search.toLowerCase());
+
+			const isAggregatorChosen = aggregator === columnName;
+
+			if (isAggregatorExist || aggregator === '') {
+				newOptions = aggregatorOptions;
+			}
+
+			if ((isAggregatorChosen && op === '') || op) {
+				const filteredOperators = numberOperators.filter((num) =>
+					num.toLowerCase().includes(op.toLowerCase()),
+				);
+
+				newOptions = filteredOperators.map((opt) => ({
+					label: `${columnName} ${opt} ${value}`,
+					value: `${columnName} ${opt} ${value}`,
+				}));
+			}
+
+			setOptions(newOptions);
```

Comment:
can we make a helper function for this one

Reply:
This function has a specific case for generating options, so i don't think it makes sense to create a separated function for it and use only in one place.

Reply:
i know we are using in one place only but how you would write unit test case for same ?

Reply:
I've already written unit test for this component and i wrote the expected result when we generate options after click. For component we need to check if the render is correct.